src/views/components/create-item-form.jsx

```diff
@@ -21,13 +22,18 @@ export default class CreateItemForm extends ListenerComponent {
     changeModel(e) {
         this.setState({
             modelAddress: e.target.value,
-            models: this.state.models
         });
     }
 
     submit(e) {
         e.preventDefault();
         createItem(this.state.modelAddress);
+
+        searchModel(this.state.modelAddress);
+        let newItemType = ModelStore.getModel();
```

Comment:
This happens immediately after the previous line, but the store won't be updated until after the async call to the server is complete.